crates/bevy_animation/src/animatable.rs

```diff
@@ -192,3 +187,157 @@ impl Animatable for Quat {
+            };
+            (*start_keyframe).clone()
+        }
+
+        Interpolation::Linear => {
+            let (Some(start_keyframe), Some(end_keyframe)) = (
+                keyframes.get_keyframe(step_start),
+                keyframes.get_keyframe(step_start + 1),
+            ) else {
+                return Err(AnimationEvaluationError::KeyframeNotPresent);
+            };
+
+            T::interpolate(start_keyframe, end_keyframe, time)
+        }
+
+        Interpolation::CubicSpline => {
+            let (
+                Some(start_keyframe),
+                Some(start_tangent_keyframe),
+                Some(end_tangent_keyframe),
+                Some(end_keyframe),
+            ) = (
+                keyframes.get_keyframe(step_start * 3 + 1),
+                keyframes.get_keyframe(step_start * 3 + 2),
+                keyframes.get_keyframe(step_start * 3 + 3),
+                keyframes.get_keyframe(step_start * 3 + 4),
+            )
+            else {
+                return Err(AnimationEvaluationError::KeyframeNotPresent);
+            };
+
+            interpolate_with_cubic_bezier(
+                start_keyframe,
+                start_tangent_keyframe,
+                end_tangent_keyframe,
+                end_keyframe,
+                time,
+                duration,
+            )
+        }
+    };
+
+    *dest = T::interpolate(dest, &value, weight);
+
+    Ok(())
+}
+
+/// Evaluates a cubic Bézier curve at a value `t`, given two endpoints and the
+/// derivatives at those endpoints.
+///
+/// The derivatives are linearly scaled by `duration`.
+fn interpolate_with_cubic_bezier<T>(p0: &T, d0: &T, d3: &T, p3: &T, t: f32, duration: f32) -> T
+where
+    T: Animatable + Clone,
+{
+    // We're given two endpoints, along with the derivatives at those endpoints,
+    // and have to evaluate the cubic Bézier curve at time t using only
+    // (additive) blending and linear interpolation.
+    //
+    // Evaluating a Bézier curve via repeated linear interpolation when the
+    // control points are known is straightforward via [de Casteljau
+    // subdivision]. So the only remaining problem is to get the two off-curve
+    // control points. The [derivative of the cubic Bézier curve] is:
+    //
+    //      B′(t) = 3(1 - t)²(P₁ - P₀) + 6(1 - t)t(P₂ - P₁) + 3t²(P₃ - P₂)
+    //
+    // Setting t = 0 and t = 1 and solving gives us:
+    //
+    //      P₁ = P₀ + B′(0) / 3
+    //      P₂ = P₃ - B′(1) / 3
+    //
+    // These P₁ and P₂ formulas can be expressed as additive blends.
+    //
+    // So, to sum up, first we calculate the off-curve control points via
+    // additive blending, and then we use repeated linear interpolation to
+    // evaluate the curve.
+    //
+    // [de Casteljau subdivision]: https://en.wikipedia.org/wiki/De_Casteljau%27s_algorithm
+    // [derivative of the cubic Bézier curve]: https://en.wikipedia.org/wiki/B%C3%A9zier_curve#Cubic_B%C3%A9zier_curves
```

Comment:
I don't really see a good way to unless we move Animatable into bevy_math, which doesn't seem like a good place for it.

To be honest, I'm not a big fan of the way that cubic Béziers are stored: I'd prefer that cubic interpolation specify the control points explicitly instead of indirectly specifying them via derivatives. Then we could replace this function with just de Casteljau subdivision, which would be both simpler and faster. But I figured that should be a follow-up if we decide to do it.

Reply:
Agreed. Feel free to mark this as resolved!